uxarray/grid/geometry.py

```diff
@@ -25,13 +25,22 @@ def _grid_to_polygons(grid, correct_antimeridian_polygons=True):
     import antimeridian
     from shapely import polygons as Polygons
 
+    # Mask out invalid faces
```

Comment:
There are some cells that, when they have a point, or an edge, or multiple, that touch the land mask, they are masked out inside the connectivity

`[a, b, c, d, e]`

If for example, `b` is on the land mask, the resulting connectivity array would be something like

`[a, -1, c, d, e]`

Theres no good way to fix this with any functionality that we have right now, so for now masking is the best approach I could think of (other than straight up removing them, but this would cause issues with the data variables not lining up to the grid dimensions)

Reply:
@rajeeja, to get into this a bit more, for an ocean mesh, the primal mesh should not have invalid faces.  But the dual mesh often has vertices (in MPAS parlance) that have only 1 or 2 valid neighboring cells because the other cell or cells are land.  You can see examples of this in the plot I did above where the dual mesh has a point or a line instead of a triangle.

Reply:
> @rajeeja, to get into this a bit more, for an ocean mesh, the primal mesh should not have invalid faces. But the dual mesh often has vertices (in MPAS parlance) that have only 1 or 2 valid neighboring cells because the other cell or cells are land. You can see examples of this in the plot I did above where the dual mesh has a point or a line instead of a triangle.

This is consistent with what I have encountered.

Reply:
By the way, I wonder if this warning:
```
UserWarning: One or more invalid face encountered in Mesh2_face_nodes. This may be due to parsing cells that have one or more node excluded for the connectivity array. Refer to Grid.Mesh2_face_maskto see which faces are being masked out
```
should be suppressed somehow or removed for MPAS meshes.  This is not a problem, it's just a property of the dual mesh for MPAS meshes that don't cover the full sphere like the ocean mesh we are working with here.

Reply:
There's an expectation that every entry in `Mesh2_face_nodes` would yeild at least a triangle from the connectivity information provided there. With the current parsing of the dual mesh containing either a point or line, we would not want those to be stored within `Mesh2_face_nodes`.